docs: cleanup ADR-002 (#1087)
umee-network/umee: docs/architecture/ADR-002-supply-assets.md

# ADR 002: Supplying Assets

## Changelog

- September 13, 2021: Initial draft (@toteki)
- September 29, 2021: Update message names to reflect implementation (@toteki)
- July 2, 2022: Cleanup and simplify (@toteki)

## Status

Implemented

## Context

> One of the base functions of the `x/leverage` module is to allow users to supply assets, and earn interest on their deposits.

Author: ```suggestion
The `x/leverage` module allows users to supply and withdraw assets.
```
